soccer/Referee.cpp

```diff
@@ -110,154 +29,186 @@ static const bool CancelBallPlaceOnHalt = true;
 Referee::Referee(Context* const context)
     : stage_(NORMAL_FIRST_HALF_PRE),
       command_(HALT),
-      _running(false),
-      _context(context) {}
+      sent_time{},
+      received_time{},
+      stage_time_left{},
+      command_counter{},
+      yellow_info{},
+      blue_info{},
+      _kickDetectState{},
+      _readyBallPos{},
+      _kickTime{},
+      _mutex{},
+      _packets{},
+      _context(context),
+      prev_command_{},
+      prev_stage_{},
+      ballPlacementX{},
+      ballPlacementY{},
+      _network_thread(),
+      _asio_socket{_io_service} {}
 
 Referee::~Referee() { stop(); }
 
+void Referee::start() {
+    setupRefereeMulticast();
+    startReceive();
+}
+
 void Referee::stop() {
-    _running = false;
-    wait();
+    _asio_socket.shutdown(boost::asio::ip::udp::socket::shutdown_both);
+    _asio_socket.close();
 }
 
-void Referee::getPackets(std::vector<RefereePacket*>& packets) {
-    _mutex.lock();
+void Referee::getPackets(std::vector<RefereePacket>& packets) {
+    std::unique_lock<std::mutex> lock(_mutex);
```

Comment:
Very small and not important at all, but since you aren't locking and unlocking the `unique_lock` or doing anything fancy, you might as well use a `lock_guard`.

Reply:
sure.